components/style/gecko/wrapper.rs

```diff
@@ -236,6 +237,7 @@ impl<'ln> GeckoNode<'ln> {
         true
     }
 
+    #[inline]
     fn flattened_tree_parent(&self) -> Option<Self> {
```

Comment:
I'm not particularly convinced this method should be marked `inline`. It looks non-trivial, especially when `flattened_tree_parent_is_parent` is inlined. And it is called in more than one places. I would say let's just let compiler decide this one.

Reply:
The whole point of `flattened_tree_parent_is_parent` is avoiding the normal ool function call, and this method is _hot_ :(